discovery/eureka/eureka.go

```diff
@@ -0,0 +1,220 @@
+	for _, t := range app.Instances {
+		targetAddress := targetEndpoint(&t)
+		target := model.LabelSet{
+			model.AddressLabel:  model.LabelValue(targetAddress),
+			model.InstanceLabel: model.LabelValue(t.InstanceID),
+
+			appNameLabel:                      lv(app.Name),
+			appInstanceHostNameLabel:          lv(t.HostName),
+			appInstanceHomePageURLLabel:       lv(t.HomePageURL),
+			appInstanceStatusPageURLLabel:     lv(t.StatusPageURL),
+			appInstanceHealthCheckURLLabel:    lv(t.HealthCheckURL),
+			appInstanceIPAddrLabel:            lv(t.IPAddr),
+			appInstanceVipAddressLabel:        lv(t.VipAddress),
+			appInstanceSecureVipAddressLabel:  lv(t.SecureVipAddress),
+			appInstanceStatusLabel:            lv(t.Status),
+			appInstancePortLabel:              lv(strconv.Itoa(t.Port.Port)),
+			appInstancePortEnabledLabel:       lv(strconv.FormatBool(t.Port.Enabled)),
+			appInstanceSecurePortLabel:        lv(strconv.Itoa(t.SecurePort.Port)),
+			appInstanceSecurePortEnabledLabel: lv(strconv.FormatBool(t.SecurePort.Enabled)),
+			appInstanceCountryIDLabel:         lv(strconv.Itoa(t.CountryID)),
+			appInstanceIDLabel:                lv(t.InstanceID),
+		}
+
+		if t.DataCenterInfo != nil {
+			target[appInstanceDataCenterInfoNameLabel] = lv(t.DataCenterInfo.Name)
+
+			if t.DataCenterInfo.Metadata != nil {
+				target[appInstanceDataCenterInfoMetadataAmiLaunchIndex] = lv(t.DataCenterInfo.Metadata.AmiLaunchIndex)
+				target[appInstanceDataCenterInfoMetadataLocalHostname] = lv(t.DataCenterInfo.Metadata.LocalHostname)
+				target[appInstanceDataCenterInfoMetadataAvailabilityZone] = lv(t.DataCenterInfo.Metadata.AvailabilityZone)
+				target[appInstanceDataCenterInfoMetadataInstanceID] = lv(t.DataCenterInfo.Metadata.InstanceID)
+				target[appInstanceDataCenterInfoMetadataPublicIpv4] = lv(t.DataCenterInfo.Metadata.PublicIpv4)
+				target[appInstanceDataCenterInfoMetadataPublicHostname] = lv(t.DataCenterInfo.Metadata.PublicHostname)
+				target[appInstanceDataCenterInfoMetadataAmiManifestPath] = lv(t.DataCenterInfo.Metadata.AmiManifestPath)
+				target[appInstanceDataCenterInfoMetadataLocalIpv4] = lv(t.DataCenterInfo.Metadata.LocalIpv4)
+				target[appInstanceDataCenterInfoMetadataHostname] = lv(t.DataCenterInfo.Metadata.Hostname)
+				target[appInstanceDataCenterInfoMetadataAmiID] = lv(t.DataCenterInfo.Metadata.AmiID)
+				target[appInstanceDataCenterInfoMetadataInstanceType] = lv(t.DataCenterInfo.Metadata.InstanceType)
+			}
+		}
+
+		if t.Metadata != nil {
+			for k, v := range t.Metadata.Map {
+				ln := strutil.SanitizeLabelName(k)
+				target[model.LabelName(appInstanceMetadataPrefix+ln)] = lv(v)
+			}
+		}
+
+		targets = append(targets, target)
+	}
+	return targets
+}
+
+func lv(s string) model.LabelValue {
+	return model.LabelValue(s)
+}
+
+func targetEndpoint(instance *Instance) string {
+	var port string
+	if instance.SecurePort.Enabled && !instance.Port.Enabled {
```

Comment:
Does secure mean https?

If so, this isn't safe and we should stick to just exposing the main port.